workbench-core/infrastructure/src/workbenchCognito.ts

```diff
@@ -137,43 +140,38 @@ export class WorkbenchCognito extends Construct {
         callbackUrls: websiteUrls,
         logoutUrls: websiteUrls
       },
+      generateSecret: true,
       accessTokenValidity: props.accessTokenValidity,
       idTokenValidity: props.idTokenValidity,
       refreshTokenValidity: props.refreshTokenValidity
     };
-    const userPoolClientProps = merge(userPoolClientDefaults, tempUserPoolClientProps);
-    this.userPoolClient = new UserPoolClient(this, 'WorkbenchUserPoolClient', {
+    const userPoolClientProps = merge(cloneDeep(userPoolClientDefaults), tempUserPoolClientProps);
+    this.webUiUserPoolClient = new UserPoolClient(this, 'WorkbenchUserPoolClient-webUi', {
       ...userPoolClientProps,
       userPool: this.userPool,
-      userPoolClientName
-    });
-    this.userPool.identityProviders.forEach((provider) => this.userPoolClient.node.addDependency(provider));
-
-    const describeCognitoUserPoolClient = new AwsCustomResource(this, 'DescribeCognitoUserPoolClient', {
-      resourceType: 'Custom::DescribeCognitoUserPoolClient',
-      onCreate: {
-        region: Stack.of(this).region,
-        service: 'CognitoIdentityServiceProvider',
-        action: 'describeUserPoolClient',
-        parameters: {
-          UserPoolId: this.userPool.userPoolId,
-          ClientId: this.userPoolClient.userPoolClientId
-        },
-        physicalResourceId: PhysicalResourceId.of(this.userPoolClient.userPoolClientId)
-      },
-      policy: AwsCustomResourcePolicy.fromSdkCalls({
-        resources: [this.userPool.userPoolArn]
-      }),
-      installLatestAwsSdk: true
+      userPoolClientName: webUiUserPoolClientName
     });
 
-    const userPoolClientSecret = describeCognitoUserPoolClient.getResponseField(
-      'UserPoolClient.ClientSecret'
-    );
+    const tempIntegrationTestUserPoolClientProps: UserPoolClientOptions = {
+      authFlows: {
+        adminUserPassword: true
+      }
+    };
+    merge(userPoolClientProps, tempIntegrationTestUserPoolClientProps);
+    this.programmaticAccessUserPoolClient = new UserPoolClient(this, 'WorkbenchUserPoolClient-iTest', {
+      ...userPoolClientProps,
+      userPool: this.userPool,
+      userPoolClientName: programmaticAccessUserPoolName
+    });
+    this.userPool.identityProviders.forEach((provider) => {
+      this.webUiUserPoolClient.node.addDependency(provider);
+      this.programmaticAccessUserPoolClient.node.addDependency(provider);
```

Comment:
I'm not 100% sure but I think the identity providers will only work through the hosted UI, so not necessary for the programmatic access client. I would double check though.

Reply:
From what I'm reading here that's correct so I'm going to remove it.